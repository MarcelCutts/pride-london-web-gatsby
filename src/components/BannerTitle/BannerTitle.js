import React from 'react'
import styled from 'styled-components'

const BannerTitle = styled.h1`
  display: block;
  color: #2d2f7f;
  font-size: 28px;
  margin-bottom: 0;

  & span {
    background-color: #ffffff;
    display: inline-block;
    padding: 0 10px;
  }
`

const StyledBannerTitle = ({ children }) => (
  <BannerTitle>
    <span>{children}</span>
  </BannerTitle>
)

export default StyledBannerTitle
