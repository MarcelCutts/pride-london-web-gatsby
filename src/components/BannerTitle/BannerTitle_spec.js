import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow } from 'enzyme'

import BannerTitle from '../BannerTitle/BannerTitle'

describe('BannerTitle', () => {
  it('should render', () => {
    const tree = renderer.create(<BannerTitle />)
    expect(tree).toMatchSnapshot()
  })

  it('should render a <span />', () => {
    const wrapper = shallow(<BannerTitle />)
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('should render child text ', () => {
    const text = 'Here is a test title!'
    const wrapper = shallow(<BannerTitle>{text}</BannerTitle>)
    expect(wrapper.find('span').text()).toBe(text)
  })
})
