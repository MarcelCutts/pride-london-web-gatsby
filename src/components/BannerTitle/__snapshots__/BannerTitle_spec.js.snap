// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerTitle should render 1`] = `
.c0 {
  display: block;
  color: #2d2f7f;
  font-size: 28px;
  margin-bottom: 0;
  padding: 0 10px;
}

.c0 span {
  background-color: #ffffff;
}

<h1
  className="c0"
>
  <span />
</h1>
`;
